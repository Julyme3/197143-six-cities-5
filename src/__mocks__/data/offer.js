/* eslint-disable */
export const offersFromAPi = [
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 1,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
    host: {
      id: 25,
      name: `Angelina`,
      is_pro: true,
      avatar_url: `img/avatar-angelina.jpg`,
    },
    id: 2,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.957361,
      longitude: 6.9509739999999995,
      zoom: 16,
    },
    max_adults: 4,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    price: 493,
    rating: 3.8,
    title: `The house among olive`,
    type: `house`,
  },
];

export const offersAdapteToClient = [
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 2,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  },
];

export const offerFromApi = {
  bedrooms: 3,
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13,
    },
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Laptop friendly workspace`, `Washer`, `Towels`, `Fridge`, `Air conditioning`, `Baby seat`],
  host: {
    id: 25,
    name: `Angelina`,
    is_pro: true,
    avatar_url: `img/avatar-angelina.jpg`,
  },
  id: 1,
  images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
  is_favorite: false,
  is_premium: false,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16,
  },
  max_adults: 4,
  preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
  price: 493,
  rating: 3.8,
  title: `The house among olive`,
  type: `house`,
};

export const fullOfferAdaptedToClient = {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `main`,
  };

export const commentsFromApi = [
  {
    comment: "The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.",
    date: "2020-11-05T13:38:44.753Z",
    id: 1,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg",
      id: 16,
      is_pro: true,
      name: "Mollie",
    },
  },
  {
    comment: "I stayed here for one night and it was an unpleasant experience.",
    date: "2020-11-05T13:38:44.753Z",
    id: 2,
    rating: 4,
    user: {
      avatar_url: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
      id: 19,
      is_pro: false,
      name: "Christina",
    },
  },
];

export const commentsAdaptedToCLient = [
  {
    id: 3,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
    name: "Isaac",
    raiting: 5,
    text: "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
    dateFormated: "November 2020",
  },
  {
    id: 2,
    date: "2020-11-05T13:38:44.753Z",
    src: "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
    name: "Christina",
    raiting: 4,
    text: "I stayed here for one night and it was an unpleasant experience.",
    dateFormated: "November 2020",
  },
];

export const favoriteOffers = {
  "Amsterdam": [
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 1,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
    {
      city: {
        location: [50.938361, 6.959974],
        name: `Cologne`,
        zoom: 13,
      },
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      type: `house`,
      price: 493,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      id: 2,
      src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      isPremium: false,
      name: `The house among olive `,
      raiting: 3.8,
      isBookmark: false,
      coords: [50.957361, 6.9509739999999995],
      details: {
        bedrooms: 3,
        adults: 4,
        insideItems: [`Laptop friendly workspace`, `Washer`],
        description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      },
      user: {
        src: `img/avatar-angelina.jpg`,
        id: 25,
        name: `Angelina`,
        isSuper: true,
      },
      typeComponent: `main`,
    },
  ],
};
