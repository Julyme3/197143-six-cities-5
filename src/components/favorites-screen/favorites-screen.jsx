import React, {PureComponent} from "react";
import {Link} from "react-router-dom";
import {FavoriteOffers} from "../../types";
import PropTypes from "prop-types";
import Footer from "../footer/footer";
import MainLayout from "../../layouts/main-layout/main-layout";
import {connect} from "react-redux";
import {fetchFavoritesAction} from "../../store/api-actions";
import {getFavorites, getIsLoading} from "../../store/reducers/favorites/selectors";
import FavoriteList from "../favorite-list/favorite-list";
import FavoritesEmpty from "../favorites-empty/favorites-empty";
import {setStartLoadingAction} from "../../store/actions";
import withActiveItem from "../../hocs/with-active-item/with-active-item";

const FavoriteListWrapped = withActiveItem(FavoriteList);
class FavoritesScreen extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.startLoading();
    this.props.fetchFavorite();
  }

  render() {
    const {favoriteOffers, isLoading} = this.props;
    const isEmpty = Object.keys(favoriteOffers).length === 0 && !isLoading;
    return (
      <>
        <MainLayout
          classNameWrap={`page__main--favorites ${isEmpty && `page__main--favorites-empty` || ``}`}
          className={`${isEmpty && `page--favorites-empty` || ``}`}
        >
          {!isEmpty ?
            <div className="page__favorites-container container">
              <section className="favorites">
                <h1 className="favorites__title">Saved listing</h1>
                <ul className="favorites__list">
                  {Object.keys(favoriteOffers)
                    .map((city) => {
                      return (
                        favoriteOffers[city].length ? <li className="favorites__locations-items" key={city}>
                          <div className="favorites__locations locations locations--current">
                            <div className="locations__item">
                              <Link className="locations__item-link" to="/">
                                <span>{city}</span>
                              </Link>
                            </div>
                          </div>
                          <FavoriteListWrapped
                            list={favoriteOffers[city]}
                          />
                        </li>
                          : null
                      );
                    })}
                </ul>
              </section>
            </div>
            :
            <FavoritesEmpty />
          }
        </MainLayout>
        <Footer />
      </>
    );
  }
}

FavoritesScreen.propTypes = {
  fetchFavorite: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  startLoading: PropTypes.func,
  favoriteOffers: FavoriteOffers,
};

const mapStateToProps = (state) => ({
  favoriteOffers: getFavorites(state),
  isLoading: getIsLoading(state),
});

const mapDispatchToProps = (dispatch) => ({
  fetchFavorite() {
    dispatch(fetchFavoritesAction());
  },
  startLoading() {
    dispatch(setStartLoadingAction());
  }
});

export {FavoritesScreen};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesScreen);
