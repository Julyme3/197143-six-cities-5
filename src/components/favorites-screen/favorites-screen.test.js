import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import {FavoritesScreen} from "./favorites-screen";
import {favoriteOffers} from "../../__mocks__/data/offer";

jest.mock(`../../layouts/main-layout/main-layout`, () => `MainLayout`);
jest.mock(`../favorites-empty/favorites-empty`, () => `FavoritesEmpty`);
jest.mock(`../favorite-list/favorite-list`, () => `FavoriteList`);
jest.mock(`../footer/footer`, () => `Footer`);

const noop = () => {};

describe(`FavoritesScreen render`, () => {
  it(`FavoritesScreen empty render`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FavoritesScreen
              startLoading={noop}
              fetchFavorite={noop}
              isLoading={false}
              favoriteOffers={{}}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`FavoritesScreen with favorites item render`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <FavoritesScreen
              startLoading={noop}
              fetchFavorite={noop}
              isLoading={false}
              favoriteOffers={favoriteOffers}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
