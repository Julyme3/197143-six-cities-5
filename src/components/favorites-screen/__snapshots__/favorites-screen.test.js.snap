// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesScreen render FavoritesScreen empty render 1`] = `
Array [
  <MainLayout
    className="page--favorites-empty"
    classNameWrap="page__main--favorites page__main--favorites-empty"
  >
    <FavoritesEmpty />
  </MainLayout>,
  <Footer />,
]
`;

exports[`FavoritesScreen render FavoritesScreen with favorites item render 1`] = `
Array [
  <MainLayout
    className=""
    classNameWrap="page__main--favorites "
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="/"
                  onClick={[Function]}
                >
                  <span>
                    Amsterdam
                  </span>
                </a>
              </div>
            </div>
            <FavoriteList
              activeItem={null}
              list={
                Array [
                  Object {
                    "city": Object {
                      "location": Array [
                        50.938361,
                        6.959974,
                      ],
                      "name": "Cologne",
                      "zoom": 13,
                    },
                    "coords": Array [
                      50.957361,
                      6.9509739999999995,
                    ],
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "details": Object {
                      "adults": 4,
                      "bedrooms": 3,
                      "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                      "insideItems": Array [
                        "Laptop friendly workspace",
                        "Washer",
                      ],
                    },
                    "id": 1,
                    "isBookmark": false,
                    "isPremium": false,
                    "name": "The house among olive ",
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "price": 493,
                    "raiting": 3.8,
                    "src": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    ],
                    "type": "house",
                    "typeComponent": "main",
                    "user": Object {
                      "id": 25,
                      "isSuper": true,
                      "name": "Angelina",
                      "src": "img/avatar-angelina.jpg",
                    },
                  },
                  Object {
                    "city": Object {
                      "location": Array [
                        50.938361,
                        6.959974,
                      ],
                      "name": "Cologne",
                      "zoom": 13,
                    },
                    "coords": Array [
                      50.957361,
                      6.9509739999999995,
                    ],
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "details": Object {
                      "adults": 4,
                      "bedrooms": 3,
                      "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                      "insideItems": Array [
                        "Laptop friendly workspace",
                        "Washer",
                      ],
                    },
                    "id": 2,
                    "isBookmark": false,
                    "isPremium": false,
                    "name": "The house among olive ",
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "price": 493,
                    "raiting": 3.8,
                    "src": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    ],
                    "type": "house",
                    "typeComponent": "main",
                    "user": Object {
                      "id": 25,
                      "isSuper": true,
                      "name": "Angelina",
                      "src": "img/avatar-angelina.jpg",
                    },
                  },
                ]
              }
              onChangeActiveItem={[Function]}
            />
          </li>
        </ul>
      </section>
    </div>
  </MainLayout>,
  <Footer />,
]
`;
